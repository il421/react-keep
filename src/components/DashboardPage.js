import React from 'react';

const DashboardPage = () => (
  <div>
    DashboardPage
  </div>
);

export default DashboardPage;