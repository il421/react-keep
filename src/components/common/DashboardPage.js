import React, { useState } from 'react';

import NoteSelection from '../note/NoteSelection';
import NotesList from '../note/NotesList';
import Header from './Header';
import SideBar from './SideBar';


const DashboardPage = () => {
  let [showBar, toggleSidebarDisplay] = useState(false);

  const showSidebar = () => {
    toggleSidebarDisplay(showBar = true);
  };

  const hideSidebar = () => {
    toggleSidebarDisplay(showBar = false);
  };

  return (
    <div className="dashboard">
      <Header showSidebar={ showSidebar } />
      <NoteSelection className="content-container"/>
      <NotesList/>

      <SideBar showBar={ showBar } hideSidebar={ hideSidebar } />
    </div>
  );
};
export default DashboardPage;
