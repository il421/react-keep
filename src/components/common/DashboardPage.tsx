import React from "react";
import { ToastContainer } from "react-toastify";
import { History } from "history";
import { JustifyContent } from "../../common";
import { Header, HeaderProps } from "./Header";
import { SideBar } from "./SideBar";
import { Controllers } from "./Controllers";
import UserFormModal from "../login/UserFormModal";
import "../../styles/components/common/_dashboard.scss";
import "../../styles/components/notes/_note-modal.scss";
import {
  NoteType,
  ImageNoteFormModal,
  TextNoteFormModal,
  ListNoteFormModal,
} from "../notes";
import { FlexBox } from "../ui-components";
import { NotesList } from "../notes/NotesList";
import { connect } from "react-redux";
import { Store } from "../../store/store.types";
import { ThunkDispatch } from "redux-thunk";
import { toggleCurrentNote } from "../../actions/modals";

export interface DashboardPageProps {
  history: History;
}

interface DispatchProps {
  toggleCurrentNote: (noteType: NoteType, isOpen: boolean, id?: string) => void;
}

export type DashboardPageBaseProps = DashboardPageProps & DispatchProps;

export const DashboardPageBase: React.FunctionComponent<DashboardPageBaseProps> = ({
  history,
  toggleCurrentNote,
}): JSX.Element => {
  return (
    <FlexBox
      vertical={true}
      justifyContent={JustifyContent.spaceBetween}
      className="dashboard"
    >
      <Header />
      <NotesList
        onNoteSelected={(type: NoteType, id: string) =>
          toggleCurrentNote(type, true, id)
        }
      />

      <SideBar />
      <TextNoteFormModal history={history} />
      <ListNoteFormModal history={history} />
      <ImageNoteFormModal history={history} />

      <UserFormModal />
      <Controllers
        isMobile
        openDialog={(type: NoteType) => toggleCurrentNote(type, true)}
      />

      <ToastContainer autoClose={2000} />
    </FlexBox>
  );
};

const mapDispatchToProps = (
  dispatch: ThunkDispatch<{}, {}, any>
): DispatchProps => ({
  toggleCurrentNote: (noteType: NoteType, isOpen: boolean, id?: string) =>
    dispatch(toggleCurrentNote(noteType, isOpen, id)),
});

export const DashboardPage = connect<DispatchProps, HeaderProps, Store>(
  undefined,
  mapDispatchToProps
)(DashboardPageBase);
