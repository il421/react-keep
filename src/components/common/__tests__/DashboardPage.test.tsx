import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { DashboardPageBase } from "../DashboardPage";
import { createBrowserHistory, History } from "history";

export const history: History = createBrowserHistory();

test("should render DashboardPage correctly", () => {
  const wrapper: ShallowWrapper<typeof DashboardPageBase, any> = shallow<
    typeof DashboardPageBase
  >(<DashboardPageBase history={history} toggleCurrentNote={jest.fn()} />);
  expect(wrapper.debug()).toMatchSnapshot();
});
