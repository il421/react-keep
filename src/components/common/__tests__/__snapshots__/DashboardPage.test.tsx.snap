// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render DashboardPage correctly 1`] = `
"<FlexBox vertical={true} justifyContent=\\"space-between\\" className=\\"dashboard\\">
  <Connect(HeaderBase) />
  <Connect(NotesListBase) onNoteSelected={[Function: onNoteSelected]} />
  <Connect(SideBarBase) />
  <TextNoteFormModal history={{...}} />
  <ListNoteFormModal history={{...}} />
  <ImageNoteFormModal history={{...}} />
  <Connect(UserFormModal) />
  <Controllers isMobile={true} openDialog={[Function: openDialog]} />
  <ToastContainer autoClose={2000} position=\\"top-right\\" transition={[Function: Animation]} rtl={false} hideProgressBar={false} closeButton={{...}} pauseOnHover={true} pauseOnFocusLoss={true} closeOnClick={true} newestOnTop={false} draggable={true} draggablePercent={80} className={{...}} style={{...}} toastClassName={{...}} bodyClassName={{...}} progressClassName={{...}} progressStyle={{...}} role=\\"alert\\" />
</FlexBox>"
`;
